### ISSUE TITLE
2x PS3 controller - one config only after relaunching fceux

### ISSUE THREAD

**Issue by radzix88** [TASVideos/fceux#398]
I have 2 separate PS3 controllers (they have different MAC address and name in BT manager). Fceux recognize them as "PS3 Controller" (with the same GUID). Both works fine with fceux, but only one config is saved in file (depends of which one was changed last). After exit and relaunching fceux both controllers has the same config and I have to change it manually every time.
![1](https://user-images.githubusercontent.com/47605609/130442593-69da05d2-a7f7-463b-b8f0-35247bddfc5c.jpg)
![2](https://user-images.githubusercontent.com/47605609/130443311-d5953655-3645-4f9f-8064-2a47ccebdab7.jpg)

**Comment by mjbudd77** [comment]
I don't think this is a bug in fceux. If the GUID is the same between the controllers, then there is no way to tell them apart. This more seems like an inconsistency between two different controller manufacturers.

**Comment by IPeluchito** [comment]
In my case, when I restart Windows 11 the emulator changes the identifier JS 0,1... by JS 255 and I have to reconfigure the controls again...
I have Dualshock 4 and use it via bluetooth with DS4Windows

**Comment by zeromus** [comment]
@mjbudd77 are you sure guids are _supposed_ to be different for different controller instances of the _same_ controller type? I'm not so sure.... you would tell them apart by the enumeration order or other integer ID or handle given to you by the joystick API youre using

**Comment by mjbudd77** [comment]
@zeromus , SDL maintains a database of default controller button mappings based on GUID. That is what my assumptions are based off of. I have seen that SDL added a function to obtain a controller serial number in later versions. The serial number may allow for these controllers to be identified separately. It would require changes to the Qt gui load/save button mapping code and maybe an SDL2 version upgrade.

**Comment by zeromus** [comment]
ok, seems like that will be needed to fix that, then.
